src/pages/index.js

```diff
@@ -164,39 +179,93 @@ profileEditButton.addEventListener("click", (evt) => {
 
 editForm.addEventListener("submit", handleEditFormSubmit);
 
-initialCards.forEach((card) => {
-  const cardElement = getCardElement(card);
-  cardsList.prepend(cardElement);
-});
-
 // New Post Modal elements
 const newPostButton = document.querySelector(".profile__add-btn");
 const newPostModal = document.querySelector("#add-card-modal");
-const newPostForm = document.forms["add-card-form"];
-
-const newPostLinkInput = newPostModal.querySelector("#add-card-link-input");
-const newPostCaptionInput = newPostModal.querySelector(
+const newPostForm = document.querySelector("#add-card-form");
+const newPostLinkInput = newPostForm.querySelector("#add-card-link-input");
+const newPostCaptionInput = newPostForm.querySelector(
   "#add-card-caption-input"
 );
 
 // For New Post Modal functions
-newPostButton.addEventListener("click", (evt) => {
+newPostButton.addEventListener("click", () => {
   openModal(newPostModal);
 });
 
 newPostForm.addEventListener("submit", handleNewFormSubmit);
 
 function handleNewFormSubmit(evt) {
   evt.preventDefault();
-  const inputValues = {
-    name: newPostCaptionInput.value,
-    link: newPostLinkInput.value,
-  };
-  const cardEl = getCardElement(inputValues);
-  cardsList.prepend(cardEl);
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+
+  api
+    .addNewCard({
+      name: newPostCaptionInput.value,
+      link: newPostLinkInput.value,
+    })
+    .then((data) => {
+      const cardEl = getCardElement(data);
+      cardsList.prepend(cardEl);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, true);
+    });
+
   evt.target.reset();
   disableButton(newPostSubmitBtn, settings);
   closeModal(newPostModal);
 }
 
+// For Avatar Modal functions
+avatarModalBtn.addEventListener("click", () => {
+  openModal(avatarModal);
+});
+
+avatarForm.addEventListener("submit", handleAvatarSubmit);
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+
+  api
+    .editAvatarInfo(avatarInput.value)
+    .then((data) => {
+      profileAvatar.src = data.avatar;
+      closeModal(avatarModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, true);
```

Comment:
`true`

* When I submit any form, the save button text is always `Saving...`, but it should get the initial button text back in `finally`  https://snipboard.io/tq59xr.jpg 

It should be `false` here


```js
      setButtonText(submitBtn, false);


```